src/assets/all_assets_request.rs

```diff
@@ -129,9 +107,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_cursor(&mut self, cursor: u32) -> &mut Self {
-        self.cursor = Some(cursor);
-        self
+    pub fn set_cursor(self, cursor: u32) -> Result<AllAssetsRequest, String> {
```

Comment:
We should add the boundaries to the documentation, so the developers know if they mess up, why their code doesn't compile.

Reply:
Moved to https://github.com/Baseflow/stellar-rust-sdk/issues/4